### PR TITLE
[SER-450] Contract serialize

### PR DESCRIPTION
**Pull request by TheButlah**
This is marked as a subtask, but I realized I could submit a PR even without the rest being implemented. So here you go, having this early should reduce review burden.

### CODE REVIEW

File: client/serialize/rust/src/serializer/mod.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+use crate::baseline::BaselineArgs;
+use crate::contract::{ContractArgs, ContractIdArgs, ContractStatesArgs};
+use crate::{c, t};
+
+pub struct Serializer<'b> {
+    fbb: FlatBufferBuilder<'static>,
+    baseline: &'b c::Baseline,
+    contracts: Vec<WIPOffset<t::Contract<'static>>>,
+    states: Vec<WIPOffset<t::State<'static>>>,
+    objects: Vec<WIPOffset<t::Object<'static>>>,
```
Comment (tomthecarrot) [inline client/serialize/rust/src/serializer/mod.rs:14]: Are these `WIP` types to be replaced in this PR's parent task?

File: client/serialize/rust/src/lib.rs [hunk 2]
```diff
@@ -4,3 +4,27 @@ pub use self::generated::tp_serialize::*;
...
+mod t {
+    pub use crate::baseline::Baseline;
+    pub use crate::contract::{Contract, ContractId, ContractStates};
+    pub use crate::object::Object;
+    pub use crate::primitive::TpPrimitive;
+    pub use crate::primitive::TpPrimitiveKind;
+    pub use crate::state::State;
+}
+
+pub const PREFIX: &str = "TPF1";
```
Comment (tomthecarrot) [inline client/serialize/rust/src/lib.rs:30]: Is the `1` in this prefix string a version code?

Comment (TheButlah) [reply client/serialize/rust/src/lib.rs:30]: Yes, teleportal flatbuffer 1. We are limited to 4 characters btw

File: client/serialize/rust/src/serializer/mod.rs (hunk 1 above)
Comment (TheButlah) [reply client/serialize/rust/src/serializer/mod.rs:14]: These types are from flatbuffer, and represent handles to tables inside the unfinished flatbuffer.

They get used to create a vector, in the `finish` method.

**Review by tomthecarrot — APPROVED** [review]

**Pull request merged**